coalib/bearlib/__init__.py

```diff
@@ -64,12 +67,27 @@ def deprecate_settings(**depr_args):
     >>> run(old='Hello!', new='Hello!')
     Hello!
 
+    Multiple deprecations can be provided for the same setting. The modifier
+    function for each deprecated setting can be given as a value in  a dict
+    where the deprecated setting is the key. A default modifier may be
+    specified at the end of the deprecated settings tuple.
+
+    >>> @deprecate_settings(new=({'old': lambda x: x + ' coala!'}, 'older'))
+    ... def run(new):
+    ...     print(new)
+    >>> run(old='Hi')
+    WARNING:root:The setting `old` is deprecated. Please use `new` instead.
+    Hi coala!
+    >>> run(older='Hi!')
+    WARNING:root:The setting `older` is deprecated. Please use `new` instead.
+    Hi!
+
     The metadata for coala has been adjusted as well:
 
     >>> list(run.__metadata__.non_optional_params.keys())
     ['new']
     >>> list(run.__metadata__.optional_params.keys())
-    ['old']
+    ['old', 'older']
 
     You cannot deprecate an already deprecated setting. Don't try. It will
     introduce non-deterministic errors in your program.
```

Comment:
I think the note is warning against the use of `@deprecate_settings(s=(s1,s2),s1=s2)` . Correct me if I am wrong.